commit 982f08775e89d3631ec871899b7270e9bdaa0463
Author: Sergei Trofimovich <siarheit@google.com>
Date:   Sat Dec 10 14:35:46 2016 +0000

    Revert "Ensure flags destined for ld are properly passed"
    
    Gentoo assumes SRC_LD_OPTS are for CC driver, not for LD:
        https://phabricator.haskell.org/D2776#82187
    
    This reverts commit f48f5a9ebf384e1e157b7b413e1d779f4289ddd2.

diff --git a/rules/distdir-way-opts.mk b/rules/distdir-way-opts.mk
index 602e6eb94d..6ae9807944 100644
--- a/rules/distdir-way-opts.mk
+++ b/rules/distdir-way-opts.mk
@@ -209,3 +209,3 @@ $1_$2_$3_ALL_LD_OPTS = \
 $1_$2_$3_GHC_LD_OPTS = \
- $$(addprefix -optl-Wl, $$($1_$2_$3_ALL_LD_OPTS)) \
+ $$(addprefix -optl, $$($1_$2_$3_ALL_LD_OPTS)) \
  $$($1_$2_$3_MOST_HC_OPTS)
